src/test/java/org/javarosa/core/model/DynamicSelectUpdateTest.java

```diff
@@ -0,0 +1,323 @@
+                                t("value"),
+                                t("label")),
+                            t("select"),
+                            t("filter"))))),
+            body(
+                repeat("/data/repeat",
+                    input("value"),
+                    input("label")),
+                select1Dynamic("/data/select", "../repeat" + (!predicate.isEmpty() ? "[" + predicate + "]" : "")),
+                input("filter")
+            )
+        );
+    }
+    //endregion
+
+    //region Select in repeat
+    // When a dynamic select is in a repeat, the itemsets for all repeat instances are represented by the same ItemsetBinding.
+    @Test
+    public void selectInRepeat_withPredicate_evaluatesChoiceListForEachRepeatInstance() throws Exception {
+        Scenario scenario = Scenario.init("Select from repeat", html(
+            head(
+                title("Select in repeat"),
+                model(
+                    mainInstance(
+                        t("data id=\"repeat-select\"",
+                            t("repeat",
+                                t("select"),
+                                t("filter")))),
+
+                    instance("choices",
+                        item("a", "A"),
+                        item("aa", "AA"),
+                        item("b", "B"),
+                        item("bb", "BB"))),
+                body(
+                    repeat("/data/repeat",
+                        select1Dynamic("/data/repeat/select", "instance('choices')/root/item[starts-with(value,current()/../filter)]"),
+                        input("filter"))
+                )
+            )));
+
+        scenario.answer("/data/repeat[0]/filter", "a");
+        scenario.answer("/data/repeat[1]/filter", "a");
+        assertThat(scenario.choicesOf("/data/repeat[0]/select").size(), is(2));
+        assertThat(scenario.choicesOf("/data/repeat[1]/select").size(), is(2));
+
+        scenario.answer("/data/repeat[1]/filter", "bb");
+        assertThat(scenario.choicesOf("/data/repeat[0]/select").size(), is(2));
+        assertThat(scenario.choicesOf("/data/repeat[1]/select").size(), is(1));
+    }
+    //endregion
+
+    //region Multi-language
+    @Test
+    public void multilanguage() throws Exception {
+        Scenario scenario = Scenario.init("Select from repeat", html(
+            head(
+                title("Multilingual dynamic select"),
+                model(
+                    t("itext",
+                        t("translation lang=\"fr\"",
+                            t("text id=\"choices-0\"",
+                                t("value", "A (fr)")),
+                            t("text id=\"choices-1\"",
+                                t("value", "B (fr)")),
+                            t("text id=\"choices-2\"",
+                                t("value", "C (fr)"))
+                        ),
+                        t("translation lang=\"en\"",
```

Comment:
On my phone, filling out this form, the French versions of the strings are presented to me, unless I modify the form, switching the order of the two `translation` elements. Is this expected?

Reply:
Yes, see [the spec](https://getodk.github.io/xforms-spec/#languages): "A default=”” attribute can be added to a <translation> to make it the default language, otherwise the first listed is used as the default."